Compilation error with clang 4.0 / sincos

(moved from https://sourceforge.net/p/gnudatalanguage/bugs/733/)
As reported on SF by Thierry Thomas:

Trying to compile GDL v. 0.9.8 on FreeBSD 11.1 (with clang 4.0), it produces the following error:
```
In file included from /wrkdirs/usr/ports/science/gnudatalanguage/work/gdl-0.9.8/src/triangulation.cpp:27:
/wrkdirs/usr/ports/science/gnudatalanguage/work/gdl-0.9.8/src/ssrfpack.c:2584:1: error: use of undeclared identifier 'sincos'
sincos (*plat, &p[2], &cos_plat);    sincos (*plon, &p[1], &p[0]);    p[0] *= cos_plat;    p[1] *= cos_plat;
^
/wrkdirs/usr/ports/science/gnudatalanguage/work/gdl-0.9.8/src/ssrfpack.c:2584:38: error: use of undeclared identifier 'sincos'
sincos (*plat, &p[2], &cos_plat);    sincos (*plon, &p[1], &p[0]);    p[0] *= cos_plat;    p[1] *= cos_plat;
```
With clang 5.0, everything is OK.

--- a/src/ssrfpack.c
+++ b/src/ssrfpack.c
@@ -7,6 +7,10 @@
 #define sincosf(x, s, c) __sincosf(x, s, c)
 #endif
 
+#ifdef __FreeBSD__
+#  define sincos(x,s,c) (*s = sin(x), *c = cos(x))
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif

